docs/master/guides/file-uploads.md

```diff
@@ -0,0 +1,59 @@
+# File Uploads
+
+Lighthouse conforms to the [graphql-multipart-request-spec](https://github.com/jaydenseric/graphql-multipart-request-spec)
+which use multipart/form-requests to transfer files in the same request as the query.
+
+This allows for file uploads through mutations.
+
+## Preparing your schema
+
+In order to accept file uploads, you must add the `Upload` scalar to your schema.
+
+```graphql
+scalar Upload @scalar(class: "Nuwave\\Lighthouse\\Schema\\Types\\Scalars\\Upload")
+```
+
+Once the scalar is added, you can add it to a mutation.
+
+````graphql
+type Mutation {
+    addProfilePicture(file: Upload!): ProfilePicture @field(resolver: "App\\GraphQL\\Mutations\\Models\\ProfilePictureMutator@add")
+}
+````
+
+## Uploading a file
```

Comment:
Maybe add a `CURL` example or something similar, as that might be easier for people to understand.